public-docs/creating-test-data.md

```diff
@@ -146,3 +166,18 @@ The `mockTags` output returns an object with custom data like this:
  }
 ]
 ```
+
+Iterators can be passed from one `makeMany` method into another `makeMany` method. This example creates many CatalogItems with CatalogProducts with a specified `product._id` value:
+
+```
+const mockCatalogItemsWithFeaturedProducts = Factory.Catalog.makeMany(30, {
+  product: (iterator) => Factory.CatalogProduct.makeOne({
+    _id: (iterator + 100).toString(),
+    isDeleted: false,
+    isVisible: true,
+    tagIds: [mockTagWithFeatured._id],
+    shopId: internalShopId
+  }),
```

Comment:
`index` is a better word than `iterator` for the passed-in variable. Usually `iterator` is used to describe a _function_ that is called multiple times. But what is passed in this case is the array index. So this whole function _is_ the "iterator":

```js
(index) => Factory.CatalogProduct.makeOne({
    _id: (index + 100).toString(),
    isDeleted: false,
    isVisible: true,
    tagIds: [mockTagWithFeatured._id],
    shopId: internalShopId
  })
```

And the iterator _receives_ the index as its first argument.

It's probably easier to see what is going on if it is separated out:

```js
function makeMockProduct(index) {
  const productId = (index + 100).toString();

  return Factory.CatalogProduct.makeOne({
    _id: productId,
    isDeleted: false,
    isVisible: true,
    tagIds: [mockTagWithFeatured._id],
    shopId: internalShopId
  });
}

const mockCatalogItemsWithFeaturedProducts = Factory.Catalog.makeMany(30, {
  product: makeMockProduct,
  shopId: internalShopId
});
```

So if any property value is a function, the `Factory` package will call that function, passing in the current `index`, and use the return value as the property value.